### PR TITLE
Migrate Geometry to new `PoolDBOutputService` methods

### PR DESCRIPTION
**Pull request by francescobrivio**
#### PR description:
Part of cms-AlCaDB/AlCaTools#28.
Migrated:
 - several plugins in ` CondTools/Geometry/plugins`
 - Geometry/DTGeometry/test/DTRecoIdealDBLoader.cc

#### PR validation:
Code compiles and `scramv1 b runtests` runs fine.

#### Backport:
N/A

### CODE REVIEW

**Comment by francescobrivio** [comment]
@cmsbuild please test

**Comment by francescobrivio** [comment]
@qliphy @perrotta @smuzaffar the relvals for this PR are in pending state for more than 8 hours, are they stuck? is there a way to restart them?

**Comment by smuzaffar** [comment]
@francescobrivio , there was glitch while cloning from github[a]. I have restarted the relval job and it is running now https://cmssdt.cern.ch/jenkins/job/ib-run-pr-relvals/12904/console

[a]
```
+ git clone https://github.com/cms-sw/cms-bot
Cloning into 'cms-bot'...
error: RPC failed; result=22, HTTP code = 503
fatal: The remote end hung up unexpectedly
```

**Comment by francescobrivio** [comment]
thanks a lot @smuzaffar!

**Comment by ggovi** [comment]
+db

File: CondTools/Geometry/plugins/PGeometricDetBuilder.cc [hunk 1]
```diff
@@ -123,8 +123,7 @@ void PGeometricDetBuilder::beginRun(const edm::Run&, edm::EventSetup const& es)
   }
   std::vector<const GeometricDet*> modules = tracker->deepComponents();
```
Comment (cvuosalo) [inline CondTools/Geometry/plugins/PGeometricDetBuilder.cc:124]: As far as I can tell, this line is useless. Am I missing something, or can it be deleted?

**Comment by cvuosalo** [comment]
@francescobrivio Thanks for doing this PR. The unit tests do not exercise any of the code changed in this PR. I can do the tests myself and put the results in this thread.

File: CondTools/Geometry/plugins/PGeometricDetBuilder.cc (hunk 1 above)
Comment (francescobrivio) [reply CondTools/Geometry/plugins/PGeometricDetBuilder.cc]: I deleted this line in 0dbe44d

**Comment by francescobrivio** [comment]
> @francescobrivio Thanks for doing this PR. The unit tests do not exercise any of the code changed in this PR. I can do the tests myself and put the results in this thread.

@cvuosalo yes please if you can do the tests and report them here it would be great! thanks! I'll re-start the jenikins tests in the meanwhile.

**Comment by francescobrivio** [comment]
@cmsbuild please test

**Comment by cvuosalo** [comment]
+1

**Comment by cvuosalo** [comment]
I tested the changes, and they work fine. The following files will be tested when the Phase 2 DD4hep DB loader is implemented:

` CondTools/Geometry/plugins/PHGCalParametersDBBuilder.cc`
` CondTools/Geometry/plugins/PTrackerAdditionalParametersPerDetDBBuilder.cc`

These two files are obsolete and probably not used, but they can be kept for now:

` CondTools/Geometry/plugins/ME0RecoIdealDBLoader.cc`
` Geometry/DTGeometry/test/DTRecoIdealDBLoader.cc`

**Comment by tvami** [comment]
+db
 * tests pass, Carl tested the real thing as well

**Comment by qliphy** [comment]
+1

**Pull request merged**